thelper/nn/resnet.py

```diff
@@ -27,66 +27,83 @@ def _make_conv2d(self, *args, **kwargs):
 
 class BasicBlock(Module):
 
-    def __init__(self, inplanes, planes, stride=1, downsample=None, coordconv=False, radius_channel=True):
+    def __init__(self, inplanes, planes, stride=1,
+                 downsample=None, coordconv=False,
+                 radius_channel=True, activation="relu"):
         super().__init__(inplanes, planes, stride, downsample, coordconv, radius_channel)
         self.conv1 = self._make_conv2d(inplanes, planes, kernel_size=3, stride=stride, padding=1, bias=False)
         self.bn1 = torch.nn.BatchNorm2d(planes)
-        self.relu = torch.nn.ReLU(inplace=True)
+        assert activation in ["relu", "leaky_relu"]
+        if activation == "relu":
+            self.activ = torch.nn.ReLU(inplace=True)
+        elif activation == "leaky_relu":
+            self.activ = torch.nn.LeakyReLU(inplace=True)
```

Comment:
I would move all duplicate activation code into another base class and add that base to all applicable modules.

Reply:
solved with a module function for now, this whole file is just hacks that should not exist